src/pages/Demos/Tabs/Tabs.tsx

```diff
@@ -0,0 +1,98 @@
+import { ReactNode } from 'react';
+import { Tabs, Badge } from '@axa-fr/react-toolkit-all';
+import Layout, { TLayout } from 'Layout';
+import LiveCode from 'shared/components/LiveCode';
+import { withEditor, useEditable, TEvent, Tknobs, EditorHeader, TReturnUseToggleEditor } from 'shared/components/Editor';
+import { TITLE_BAR, TITLE, DESIGN_SYSTEM_PATH, STORYBOOK_PATH, GITHUB_PACKAGE, NPM_NAME } from './constants';
+import knobs from './knobs.json';
+
+const INITIAL_STATE = {
+  labelFirstTab: '<i className="glyphicon glyphicon-briefcase" /> Title with left icon',
+  labelSecondTab: '<i className="glyphicon glyphicon-user" /> Title with right icon',
+  labelThirdTab: 'Title with badge <Badge classModifier="info"> 21 </Badge>',
+  contentFirstTab: 'Content first tab',
+  contentSecondTab: 'Content second tab',
+  contentThirdTab: 'Content second tab',
+  classModifierFirstTab: 'has-icon-left',
+  classModifierSecondTab: 'has-icon-right',
+  className: 'af-tabs',
+  activeIndex: '1',
+};
+
+type Props = Partial<typeof INITIAL_STATE> & {
+  onChange: (name: keyof typeof INITIAL_STATE) => (arg: TEvent) => void;
+  onChangeTab: (arg: string) => void;
+  isOpenEditor?: boolean;
+};
+
+export const code = ({
+  labelFirstTab,
+  labelSecondTab,
+  labelThirdTab,
+  contentFirstTab,
+  contentSecondTab,
+  contentThirdTab,
+  className,
+  classModifierFirstTab,
+  classModifierSecondTab,
+  activeIndex,
+  isOpenEditor,
+}: Props) => `
+    <Tabs className="${className}" onChange={onChangeTab} activeIndex="${activeIndex}">
+        <Tabs.Tab title={<>${labelFirstTab}</>} classModifier="${classModifierFirstTab}">
+            ${contentFirstTab}
+        </Tabs.Tab>
+        <Tabs.Tab title={<>${labelSecondTab}</>} classModifier="${classModifierSecondTab}" >
+            ${contentSecondTab}
+        </Tabs.Tab>
+        <Tabs.Tab title={<>${labelThirdTab}</>}>
+            ${contentThirdTab}
+        </Tabs.Tab>
+    </Tabs>
+  `;
+
+const TabsWithEditor = withEditor<Props & Partial<TReturnUseToggleEditor>>(
+  ({ openEditor, ...props }) => (
+    <>
+      <EditorHeader
+        storybookPath={STORYBOOK_PATH}
+        designSystemPath={DESIGN_SYSTEM_PATH}
+        githubPackage={GITHUB_PACKAGE}
+        npmName={NPM_NAME}
+        openEditor={openEditor}
+      />
+      <LiveCode
+        classModifier="with-editor"
+        styleLivePreview={{ textAlign: 'left' }}
+        code={code(props)}
+        scope={{
+          Tabs,
+          Badge,
+          ...props,
+        }}
+      />
+    </>
+  ),
+  knobs as unknown as Tknobs,
+);
+
+export const onChangeTabFn = (key: string, onChangeFn: (key: string) => (e: TEvent) => void) => (e: string) => onChangeFn(key)({ value: e });
+
+const TabsEditable = () => {
+  const { state, onChange } = useEditable<typeof INITIAL_STATE>({ initialState: INITIAL_STATE });
+  return <TabsWithEditor {...state} onChange={onChange} onChangeTab={onChangeTabFn('activeIndex', onChange)} />;
```

Comment:
on ne pouvait juste faire ça ?
```suggestion
  return <TabsWithEditor {...state} onChange={onChange} onChangeTab={onChange('activeIndex')} />;
```

Reply:
Non car le onChange vient du useEditable, et dans cette méthode tu fais : 
`setState(prevState => ({ ...prevState, [key]: setValueFn(e.value) }));`
Or pour les tabs, le toolkit renvoie juste un string (sur le onChange) et non un objet avec value dedans, donc pour éviter de changer le code côté onChange du useEditable, il a fallu surcharger le onChange.